### ISSUE TITLE
Controller Widget crash

### ISSUE THREAD

**Issue by thrust26** [stella-emu/stella#954]
It seems that the debugger's controller widgets get decoupled from their controllers when the GameProperties dialog is used.

Example: 

1. start Combat
2. enter debugger and check I/O tab (all controller checkboxes are unchecked)
3. exit debugger
4. press Tab to open Options dialog
5. open GameProperties dialog and leave with OK button
6. enter debugger and check I/O tab (some controller checkboxes are checked)

The problem is in step 5, there the controllers are created again. But the controller widgets are still referring to their previous controller instances.

BTW: I the controller is a SaveKey/AtariVox the debugger will crash.

**Comment by thrust26** [comment]
@sa666666 This is tricky. Do you have an idea how to fix this?

**Comment by thrust26** [comment]
@sa666666 Is this still planned for 7.0?

**Comment by sa666666** [comment]
Yes, still planned for 7.0.  When I get back to working on Stella again (soon).

**Comment by sa666666** [comment]
The dangling issue and potential crashes are fixed in cd292f18b32a2887d1648d14f6d4abe29dd08e43.  However, I'll keep this open since the larger issue is still not fixed; changing the controller _type_ outside the debugger is not yet recognized in the debugger I/O tab, so it will always show the original controller.

**Comment by thrust26** [comment]
Still better than a crash.
